homeassistant/components/yolink/services.py

```diff
@@ -28,6 +27,13 @@ async def handle_speaker_hub_play_call(service_call: ServiceCall) -> None:
         device_registry = dr.async_get(hass)
         device_entry = device_registry.async_get(service_data[ATTR_TARGET_DEVICE])
         if device_entry is not None:
+            for entry_id in device_entry.config_entries:
+                if (entry := hass.config_entries.async_get_entry(entry_id)) is None:
+                    continue
+                if entry.domain == DOMAIN:
+                    break
+            if entry is None:
+                return
             home_store = hass.data[DOMAIN][entry.entry_id]
```

Comment:
We need to check that the state of the config entry is loaded before accessing hass.data. Raise `ServiceValidationError` if it's not loaded.

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/

Reply:
Please also check the state of the config entry.